tests/compiler/test_list.py

```diff
@@ -242,8 +242,9 @@ def test_608_opt_to_list(snippetcompiler):
 b1.a = c1.a
 """
     )
-    with pytest.raises(OptionalValueException):
+    with pytest.raises(AttributeException) as exc_info:
```

Comment:
Why did this change?

Reply:
The error is now raised only when we try to fetch the value from the attribute, rather when we try to dereference it. So it is now caught and reraised my `SetAttributeHelper`.